tests/common/utils.ts

```diff
@@ -0,0 +1,180 @@
+
+    console.log(`${proofType} Setting up event listeners...`);
+    const eventResults = withAttestation
+        ? handleEventsWithAttestation(events, proofType, 'verify')
+        : handleCommonEvents(events, proofType, 'verify');
+
+    console.log(`${proofType} Transaction result received. Validating...`);
+
+    const transactionInfo: VerifyTransactionInfo = await transactionResult;
+    validateVerifyTransactionInfo(transactionInfo, proofType, withAttestation);
+    validateEventResults(eventResults, withAttestation);
+
+    if (validatePoe) {
+        await validatePoE(session, transactionInfo.attestationId!, transactionInfo.leafDigest!);
+    }
+
+    console.log(`${proofType} Closing session...`);
+    await session.close();
+
+    return { eventResults, transactionInfo };
+};
+
+
+export const performVKRegistrationAndVerification = async (
+    proofType: ProofType,
+    proof: any,
+    publicSignals: any,
+    vk: string
+): Promise<void> => {
+    const session = await zkVerifySession.start().Testnet().withAccount(process.env.SEED_PHRASE!);
+
+    console.log(`${proofType} Executing VK registration and setting up event listeners...`);
+    const { events: registerEvents, transactionResult: registerTransactionResult } = await session.registerVerificationKey()[proofType]().execute(vk);
+
+    const registerResults = handleCommonEvents(registerEvents, proofType, 'vkRegistration');
+
+    const vkTransactionInfo: VKRegistrationTransactionInfo = await registerTransactionResult;
+    validateVKRegistrationTransactionInfo(vkTransactionInfo, proofType);
+    validateEventResults(registerResults, false);
+
+    console.log(`${proofType} Executing verification using registered VK and setting up event listeners...`);
+    const { events: verifyEvents, transactionResult: verifyTransactionResult } = await session.verify()[proofType]().withRegisteredVk().execute(proof, publicSignals, vkTransactionInfo.statementHash!);
+    const verifyResults = handleCommonEvents(verifyEvents, proofType, 'verify');
+
+    const verifyTransactionInfo: VerifyTransactionInfo = await verifyTransactionResult;
+    validateVerifyTransactionInfo(verifyTransactionInfo, proofType, false);
+    validateEventResults(verifyResults, false);
+
+    console.log(`${proofType} Closing session...`);
+    await session.close();
+};
+
+
+export const validateTransactionInfo = (
+    transactionInfo: TransactionInfo,
+    expectedProofType: string
+): void => {
+    expect(transactionInfo).toBeDefined();
+    expect(transactionInfo.blockHash).not.toBeNull();
+    expect(transactionInfo.proofType).toBe(expectedProofType);
+    expect(transactionInfo.status).toBe(TransactionStatus.Finalized);
+    expect(transactionInfo.txHash).toBeDefined();
+    expect(transactionInfo.extrinsicIndex).toBeDefined();
+    expect(transactionInfo.feeInfo).toBeDefined();
+    expect(transactionInfo.weightInfo).toBeDefined();
+    expect(transactionInfo.txClass).toBeDefined();
```

Comment:
Very minor: probably not worth to perform this checks as we can assume that (if the transaction didn't error) the underlying APIs are correct and will populate these fields

Reply:
Just to be safe though, I don't like to assume :D